starkinfra/creditnote/__creditnote.py

```diff
@@ -48,10 +51,35 @@ def __init__(self, template_id, name, tax_id, nominal_amount, scheduled, invoice
         self.created = check_datetime(created)
         self.updated = check_datetime(updated)
 
+        self.payment, self.paymentType = _parse_payment(payment, paymentType)
+
 
 _resource = {"class": CreditNote, "name": "CreditNote"}
 
 
+def _parse_payment(payment, paymentType):
+    if isinstance(payment, dict):
+        try:
+            return from_api_json(*({
+                "transfer": _transfer_resource,
+            }[paymentType], payment)), paymentType
+        except KeyError:
+            return payment, paymentType
+
+    if paymentType:
+        return payment, paymentType
+
+    if isinstance(payment, Transfer):
+        return payment, "transfer"
+
+    raise Exception(
+        "payment must be either "
+        "a dictionary"
+        ", a starkinfra.creditnote.Transfer"
+        ", but not a {}".format(paymentType(payment))
```

Comment:
Acredito que essa estrutura vai quebrar se for chamada, dado que paymentType é uma string.